### ISSUE TITLE
P1: Preserve non-zero RL policy-gradient updates for small gradients

### ISSUE THREAD

**Issue by lanyusea** [lanyusea/screeps#1232]
## Context
Latest Loop A training ledger `runtime-artifacts/rl-control-loop/20260519T131517Z-training-ledger.json` reports P1 anomaly `POLICY_UPDATE_NO_PARAMETER_CHANGE`:

- Run `tencent-pg-20260518t163503z` computed a REINFORCE gradient but produced zero parameter changes because the current bounded integer-step update rounded all small deltas to 0.
- Example gradient values: `territorySignalWeight=+0.0085`, `killSignalWeight=-0.0008`, `resourceSignalWeight=+0.0037`, `riskPenalty=+0.0016`; with learning rate 0.25 and integer step=1, every update is lost.
- #924 is closed as the scorecard contract; this is a new atomic implementation/evaluation item, not a reason to reopen #924.

## Acceptance criteria
- Evaluate and implement the smallest safe change so valid small policy-gradient signals are not silently discarded: either continuous/float parameter updates, adjusted learning-rate/step rules, or explicit no-op classification with a next-card delta.
- Add tests proving a nonzero gradient produces a nonzero candidate delta when it should, and that bounds/safety constraints still hold.
- Preserve all RL safety flags (`liveEffect=false`, `officialMmoWrites=false`) and #907/#924 decision discipline.
- Produce a post-fix training/scorecard artifact or ledger evidence showing the anomaly is gone or deliberately classified with a non-silent reason.

## Evidence
- `runtime-artifacts/rl-control-loop/20260519T131517Z-training-ledger.json`
- `runtime-artifacts/tencent-cloud/batch-runs/tencent-pg-20260518t163503z/remote/runtime-artifacts/rl-training/tencent-pg-20260518t163503z.json`

## Next action
Queue after PR #1229 and the stale E1 gate repair unless it becomes the next blocker for true-gradient candidate generation.

**Comment by lanyusea** [comment]
RL steward checkpoint — 2026-05-19T14:59Z

Latest Loop A ledger `runtime-artifacts/rl-control-loop/20260519T144516Z-training-ledger.json` keeps this issue relevant and slightly reframes it:

- `trainingDidRun=true`; `envRequested=125`, `envCompleted=124`, `ticksRun=62000`, `episodes=124`.
- `policyUpdateIterations=0` across the 5 indexed training reports; ledger notes the current policy-gradient path is still fixed-vector evaluation/ranking, not true gradient updates.
- `tencent-pg-20260519t120820z` activation proof passed for territory/risk-aware seeds, but PR #1229 is still the active review gate for runtime-parameter evidence semantics.

Next action remains queued, not dispatched: after PR #1229 exits review and merges, dispatch the smallest Codex slice to make non-zero gradient/ranking signals produce a non-zero safe candidate delta or an explicit non-silent no-op classification with tests. This run did not launch a duplicate lane because Codex `proc_8459cf532cc4` is already active on PR #1229 review-fix in the same RL gate chain.

**Comment by lanyusea** [comment]
Scheduler dispatch 2026-05-20T07:33Z: started Codex implementation lane for #1232.

- Worktree: `/root/screeps-worktrees/issue-1232-small-gradients`
- Branch: `fix/issue-1232-small-gradients`
- Base: current `origin/main` / `53749732a2b3ab8c216e9276b0896149e055c3cb`
- Hermes process session: `proc_9a99cdff264a`
- OS pid: `1753920`
- Scope: preserve non-zero policy-gradient updates for small valid gradients, with focused tests and Codex-authored commit.
- Safety: prompt explicitly forbids paid Tencent compute, official deploy, secrets, and GitHub/Project mutation; controller will verify/push/open PR after exit.

Next action: monitor the Codex session; after it exits, inspect real worktree status/log/diff, run controller verification, recover commit if necessary, then create the PR.

**Comment by lanyusea** [comment]
PR #1260 merged as `91b17a2a` with the code/test fix for small policy-gradient updates.

Post-merge controller verification on `main`:
- `git diff --check HEAD^..HEAD`
- `python3 -m py_compile scripts/screeps_rl_training_runner.py scripts/test_screeps_rl_training_runner.py`
- `python3 -m pytest scripts/test_screeps_rl_training_runner.py -q` → 69 passed, 13 subtests passed

Important: PR #1260 intentionally used `Refs #1232` instead of `Fixes #1232`, so this issue remains open for the final acceptance criterion: produce post-fix training/scorecard/ledger evidence showing the anomaly is gone or deliberately classified with a non-silent reason.

Next action: inspect the next Loop A / training-run evidence on `main` after `91b17a2a`; the already-active Tencent run `tencent-pg-20260520t074004z` started from pre-PR head `53749732` and must not be counted as post-fix evidence.
